File: C++/Function/001- Factorial using Recursion.cpp
#include<iostream>
using namespace std;
int factorial(int n){
    if(n==1)
    return 1;
    else
    return n*factorial(n-1);
}
main(){
     int f;
    cout<<"Enter number:";
    cin>>f;
    cout<<"Factorial of "<<f<<" is:\t"<<factorial(f);
}

File: Java/001-Factorial using Recursion.java
import java.util.Scanner;
public class recfactorial {
    static int factorial(int n){
        if(n==1)
        return 1;
        else
        return n*factorial(n-1);
    }
    public static void main(String[] args) {
        int n, fact;
        Scanner sc=new Scanner(System.in);
        System.out.println("Enter Number: ");
        n =sc.nextInt();
        fact= factorial(n);
        System.out.println("Factorial of "+n+" is:\t"+fact);
    }}
